### PR TITLE
8315444: Convert test/jdk/tools to Classfile API

### PR DESCRIPTION
**Pull request by xqoasis**
`/test/jdk/tools/lib/tests/JImageValidator.java`, tests in `/test/jdk/tools/jlink`, and `/test/jdk/tools/jimage`, `/test/jdk/java/time/nontestng/java/time/chrono/HijrahConfigTest.java` use on com.sun.tools.classfile and should be converted to Classfile API.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315444](https://bugs.openjdk.org/browse/JDK-8315444): Convert test/jdk/tools to Classfile API (**Sub-task** - P5)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15529/head:pull/15529` \
`$ git checkout pull/15529`

Update a local copy of the PR: \
`$ git checkout pull/15529` \
`$ git pull https://git.openjdk.org/jdk.git pull/15529/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15529`

View PR using the GUI difftool: \
`$ git pr show -t 15529`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15529.diff">https://git.openjdk.org/jdk/pull/15529.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15529#issuecomment-1702950299)

### CODE REVIEW

**Comment by asotona** [comment]
It looks good, thanks.

**Comment by asotona** [comment]
Following tier2 test is affected by the change and fails:
test/jdk/java/time/nontestng/java/time/chrono/HijrahConfigTest.java

File: test/jdk/tools/lib/tests/JImageValidator.java [hunk 1]
```diff
@@ -225,10 +222,6 @@ public long getModuleLauncherExecutionTime() {
     }
 
     public static void readClass(byte[] clazz) throws IOException {
-        try (InputStream stream = new ByteArrayInputStream(clazz)) {
-            ClassFile.read(stream);
-        } catch (ConstantPoolException e) {
-            throw new IOException(e);
-        }
+        Classfile.of().parse(clazz);
```
Comment (liach) [inline test/jdk/tools/lib/tests/JImageValidator.java:225]: Notice that the old API might throw different exceptions compared to the new one; the new one is lazy, e.g. if there's broken interface entry but the interfaces are not queried, no exception is thrown.

**Comment by liach** [comment]
In addition, I don't think we should open the `classfile.impl` package to tests; this package is not meant to be exported.

File: test/jdk/tools/lib/tests/JImageValidator.java (hunk 1 above)
Comment (asotona) [reply test/jdk/tools/lib/tests/JImageValidator.java:225]: You are right, however original `ClassFile::read` also does not represent much of a validation.
Stronger verification can by applied by asserting no verify errors returned from `ClassModel::verify`.

**Review by asotona — APPROVED** [review]

**Pull request closed without merging**